### DISCUSSION TITLE
How can I configure to connect to wallets like okx wallet?

### DISCUSSION THREAD

**Discussion by percyzt** [wevm/wagmi#2676] [Q&A]
How can I configure to connect to wallets like okx wallet? I want to connect to an okx wallet: sending transactions, switching networks, But I found out that the okx wallet was actually called by the Meta Mask connector. How can I solve it?
![image](https://github.com/wagmi-dev/wagmi/assets/96770326/ee26fc8a-eff9-4eaf-87ee-f377c1d2ba13)
At this point, I clicked on the meta mask but it will be called up as an okx wallet. How can I solve this problem?

**Comment by inna0202** [comment]
helo

**Reply by percyzt** [reply]
Can you be more detailed?

**Comment by shrekuu** [comment]
@youthinn 

It's 2024-03-28. It seems that OKX Wallet does not intercept MetaMask injected `window.ethereum` object anymore.

Now you can create wagmi config like this to try to connect to MetaMask and OKX Wallet.

```tsx
const chains = [mainnet];

const { publicClient, webSocketPublicClient } = configureChains(chains, [publicProvider()]);

const config = createConfig({
  autoConnect: true,
  publicClient,
  webSocketPublicClient,
  connectors: [
    new MetaMaskConnector({ chains }),
    new InjectedConnector({
      chains,
      options: {
        name: 'OKX Wallet',
        getProvider: () => (typeof window !== 'undefined' ? window.okxwallet : undefined),
      },
    }),
  ],
});
```
Pass above config to you `WagmiProvider`.

When you want to connect to a specific one, call this only in the browser:

```tsx
import { MetaMaskConnector } from 'wagmi/connectors/metaMask';
import { InjectedConnector } from 'wagmi/connectors/injected';

// connect to MetaMask
connect({ connector: new MetaMaskConnector() });

// connect to OKX Wallet
connect({
    connector: new InjectedConnector({
      options: {
        name: 'OKX Wallet',
        getProvider: () => window.okxwallet,
      },
    }),
  });

// connect to either of the two
const res = await connect({
  connector: window.ethereum
    ? new MetaMaskConnector({ chains })
    : new InjectedConnector({
        chains,
        options: {
          name: 'OKX Wallet',
          getProvider: () => window.okxwallet,
        },
      }),
});
```

Maybe using [ConnectKit](https://docs.family.co/connectkit), [Web3Modal](https://web3modal.com/) together would make life easier.
